### ISSUE TITLE
Simulator Issues

### ISSUE THREAD

**Issue by KrishnaDwibhashi2** [glific/glific-frontend#3112]
Simulator on Glific UI always often gives different issues (which are non-replicable). There were fixes made earlier, but the issues persists.

Here are some of the commonly observed issues-
- Flow not starting (old flow still going on/ new flow not starting). It this case often keyword is sent, but there is no response.
In these cases, if we cancel & reopen simulator, then the flow gets continued
![image](https://github.com/user-attachments/assets/7ac681cc-5081-45c4-b15a-eaa993f62253)

- Input is taken incorrectly- it can be either keyword/ input given

**Comment by radhikaT4D** [comment]
@KrishnaDwibhashi2  a few questions:
1) which browser were you using. Does it happen on any browser? 
2) Also if possible try to capture a clip when this issue happens, and attach here, with a timestamp, for devs to review.

**Comment by kurund** [comment]
@KrishnaDwibhashi2 

Also wondering if this happens only to specific flows such as when webhooks are part of the flow?

**Comment by KrishnaDwibhashi2** [comment]
@kurund No, it happens in normal flows too where there are no webhooks, but just basic features like interactive messages, send message, wait for response, update contact etc

@radhikaT4D I use chrome browser- hasn't tested on other browsers, but many ngos reported it multiple times (so I am assuming its not browser specific)
I will capture it next time and share

**Comment by kurund** [comment]
@KrishnaDwibhashi2 

As mentioned during one of our calls, I do have some sort of idea for this weird behaviour. I need to investigate a bit more to confirm and then work on the fix.

**Comment by akanshaaa19** [comment]
Added some observations here
https://docs.google.com/document/d/1DHUq_Qfi2v8VCBnqMHqZFKeMpf72kvWcvPeNDeO9b3E/edit?tab=t.0
